.github/workflows/bundler.yml

```diff
@@ -31,10 +31,21 @@ jobs:
           make cluster.clean cluster.opensearch.build cluster.opensearch.start
 
       - name: Use Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 14.x
 
+      # NPM started understanding yarn.lock file starting from v7
+      - name: Update NPM
+        shell: bash
+        run: |
+          export NPM_VERSION=$(npm -v)
+          export IS_NPM_SUITABLE=$(node -p "parseInt(process.env.NPM_ROOT, 10) >= 7")
+          if [ "$IS_NPM_SUITABLE" == "false" ]; then
+            echo "NPM needs upgrading!"
+            npm i npm@7 -g
+          fi
```

Comment:
Isn't this always needed if we're on node 14.x?

Reply:
The version of NPM we get is entirely up to `actions/setup-node`. Right now they give us v6 for 10, 12, and 14. They might choose to include npm@8 later and that's why, I am not assuming anything and attempting to detect the version we have.